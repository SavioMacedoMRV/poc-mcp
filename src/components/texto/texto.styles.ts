import styled from 'styled-components'
import {HTMLAttributes, ReactNode} from 'react'
import {fontes} from 'resources/fontes'
import {cores} from 'resources/cores'

export interface TextoProps extends HTMLAttributes<HTMLSpanElement> {
  children?: ReactNode
  cor?: string
  estilo?: 'bold' | 'semibold' | 'regular'
  tamanho?: 10 | 12 | 14 | 16 | 18 | 20 | 24 | 28 | 32
  alturalinha?: 12 | 14 | 16 | 18 | 20 | 24 | 28 | 30 | 32 | 40
  peso?: 300 | 400 | 600 | 800
  alinhamento?: 'start' | 'end' | 'left' | 'right' | 'center' | 'justify'
  cursor?: 'default' | 'pointer' | 'text'
  transformar?: 'capitalize' | 'lowercase' | 'uppercase'
}

export const TextoStyled = styled.span<TextoProps>`
  font-weight: ${({peso}) => peso ?? 400};
  font-size: ${({tamanho}) => tamanho ?? 12}px;
  line-height: ${({alturalinha}) =>
    alturalinha ? alturalinha + 'px' : 'normal'};
  font-family: ${({estilo}) => (estilo ? fontes[estilo] : fontes.regular)};
  color: ${({cor}) => cor ?? cores.neutralXXDark};
  text-align: ${({alinhamento}) => alinhamento ?? 'start'};
  cursor: ${({cursor}) => cursor ?? 'text'};
  text-transform: ${({transformar}) => transformar ?? 'none'};
`
