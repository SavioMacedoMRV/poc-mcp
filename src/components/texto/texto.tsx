import React from 'react'
import {StyleSheetManager} from 'styled-components'
import {TextoStyled, TextoProps} from './texto.styles'

export const Texto = (props: Readonly<TextoProps>) => {
  return (
    <StyleSheetManager shouldForwardProp={prop => !!prop}>
      <TextoStyled {...props} />
    </StyleSheetManager>
  )
}
