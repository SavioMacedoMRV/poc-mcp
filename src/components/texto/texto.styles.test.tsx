import {render} from '@testing-library/react'
import {TextoStyled} from './texto.styles'

describe('Texto.styles', () => {
  test('TextoStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<TextoStyled />)
    expect(container).toMatchSnapshot()
  })
})
