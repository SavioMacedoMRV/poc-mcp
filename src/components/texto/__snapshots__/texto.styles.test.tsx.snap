// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texto.styles TextoStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 12px;
  line-height: normal;
  font-family: AvertaStd-Regular;
  color: #242625;
  text-align: start;
  cursor: text;
  text-transform: none;
}

<div>
  <span
    class="c0"
  />
</div>
`;
